src/I18n/Time.php

```diff
@@ -799,6 +838,100 @@ public static function parseTime($time, $format = null)
     }
 
     /**
+     * Returns the calendar type
+     * @param string|null $locale The locale name in which the date should be displayed (e.g. pt-BR)
+     * @return integer \IntlDateFormatter::TRADITIONAL|\IntlDateFormatter::GREGORIAN
+     */
+    public static function detectCalendarType($locale = null)
+    {
+        if (preg_match('/@calendar=(japanese|buddhist|chinese|persian|indian|islamic|hebrew|coptic|ethiopic)/', $locale)) {
+            return \IntlDateFormatter::TRADITIONAL;
+        }
+        return \IntlDateFormatter::GREGORIAN;
+    }
+
+    /**
+     * Sets the default locale used when need when converting date to locale.
+     *
+     * @param string $calendar set calendar name.
+     * @return void
+     */
+    public static function setDefaultCalendar($calendar)
+    {
+        static::$defaultCalendar = $calendar;
```

Comment:
Ahouldd you check that this method gets a string?  Right now other methods may misbehave if 0/false is provided.